x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -1109,9 +1109,6 @@ public AsyncSender interceptSender(AsyncSender sender) {
 
     @Override
     public UnaryOperator<RestHandler> getRestHandlerWrapper(ThreadContext threadContext) {
-        if (enabled == false) {
-            return null;
-        }
         final boolean ssl = HTTP_SSL_ENABLED.get(settings);
```

Comment:
I think we need this:
```suggestion
        final boolean ssl = enabled && HTTP_SSL_ENABLED.get(settings);
```
If security is disabled, then HTTP SSL will not be enabled either (even if the setting is true).